service/lib/agama/dbus/interfaces/locale.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2024] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "dbus"
+
+Yast.import "WFM"
+
+module Agama
+  module DBus
+    module Interfaces
+      # Mixin to define the Locale interface.
+      #
+      # @note This mixin is expected to be included in a class that inherits from {DBus::BaseObject}
+      # and it requires a #locale= method that sets the service's locale.
+      module Locale
+        include Yast::I18n
+        include Yast::Logger
+
+        LOCALE_INTERFACE = "org.opensuse.Agama1.Locale"
+
+        def self.included(base)
+          base.class_eval do
+            dbus_interface LOCALE_INTERFACE do
+              dbus_method :SetLocale, "in locale:s" do |locale|
```

Comment:
Names of locales differ, notably between Linux and browsers: `en_US.UTF-8` vs `en-us` so we should specify what the API expects or we will get subtle bugs

Reply:
In this case, it expects an `en_US.UTF-8` kind of locale. That's the only locale that the backend understands. The transformation is performed in the [installerL10n.jsx context](https://github.com/openSUSE/agama/blob/master/web/src/context/installerL10n.jsx#L109-L140).

I have added a comment to clarify that. Thanks!